core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -580,23 +580,26 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // This excludes PushJoinIntoTableScan because it must run after ReorderJoins to not undo join reordering
```

Comment:
How would `PushJoinIntoTableScan` undo `ReorderJoins`?

Reply:
Thanks, misleading comment. `PushJoinIntoTableScan` will prevent `ReorderJoins` (because JoinNode will be lost).

Will change.

Reply:
> because JoinNode will be lost

It won't be lost technically. Connector can do it's own pushdown. But yes, holistically it might be less efficient